### DISCUSSION TITLE
Total PV Generation unit measurement

### DISCUSSION THREAD

**Discussion by Elrikus** [slipx06/sunsynk-power-flow-card#481] [Ideas]
Hi! I use the total_pv_generation entity window to display information from the sensor, which shows how many kWh were produced from 1 kWp. The unit of measurement is kWh/kWp. However, this unit is not supported in the code, so the measurement data is rounded to full unity. Please add kWh/kWp with an accuracy of 2 decimal places. I'm adding screenshots to see what the difference is. Thank you! 
<img width="306" alt="Zrzut ekranu 2024-06-16 o 11 01 00" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/31111410/fcbe5cbe-ea13-4af0-a2e9-72c725f839e8">
<img width="306" alt="Zrzut ekranu 2024-06-16 o 11 02 22" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/31111410/87df3a75-c3f0-4af2-b9d5-cfda99e66ed8">

**Comment by slipx06** [comment]
Hi. I'll take a look.

**Reply by Roving-Ronin** [reply]
kWp is not an accepted unit of meaurement for device class 'Power' in HA, even for 'Energy' it isn't...?

https://developers.home-assistant.io/docs/core/entity/sensor/

**Reply by Elrikus** [reply]
I tried to bypass the system with the help of GPT, but without success. 
![Screenshot_20240619_091159](https://github.com/slipx06/sunsynk-power-flow-card/assets/31111410/bf087870-5654-4355-9d44-5f2f9a9a9371)

**Reply by Roving-Ronin** [reply]
@Elrikus 
Doesn't your integration for Deye inverter publish a 'Daily Yield' figure? If so, you just need to enter that as the sensor to use.
Alternatively if the integration only provides a 'lifetime' yield figure, you just need to use the utility_meter in HA to have this publish a Daily Yield instead.

https://www.home-assistant.io/integrations/utility_meter/

i.e.
```
utility_meter:
  deye_inverter_yield_daily:
    unique_id: deye_inverter_yield_daily
    name: Deye Inverter - Yield Daily
    source: sensor.deye_inverter_yield_lifetime
    cycle: daily
```

repeat and change to weekly / monthly / quarterly / yearly if you also want to track those amounts.

**Reply by Elrikus** [reply]
I think you mean daily production expressed in kWh. I have it. What I mean is to convert it into the amount of kWh from one kWp. This is helpful when comparing production results with other PV installations because it expresses a comparable value regardless of the size of the installation. I have created a sensor that does this, but unfortunately it cannot be shown on power_flow_card with the correct unit. If it is not possible, it is difficult to say, but it is possible that there is a way to correctly display the unit in the card.
<img width="312" alt="Zrzut ekranu 2024-06-19 o 11 18 59" src="https://github.com/slipx06/sunsynk-power-flow-card/assets/31111410/60687f4a-8425-40cb-8bdc-a64de947c766">

and here is the correct sensor:
```
      - name: "Energia_na_kWp"
        unit_of_measurement: 'kWh/kWp'
        device_class: energy
        state: >
          {% set energy_produced = states('sensor.sunsynk_day_pv_energy') | float %}
          {% set system_capacity = 9.63 %}
          {{ (energy_produced / system_capacity) | round(2) | float }}
        attributes:
          friendly_name: "Współczynnik wyprodukowanej energii/1kWp"
```

**Reply by Roving-Ronin** [reply]
Ah OK, trying to show how many kWh was produced in the day, for each 1kW of panels. Assuming 9.63 is the size of your panels in kW?

comment: this measurement/ ratio will only hold up where the panels are the same or smaller than the inverter capacity. For example in Australia we install panels at 133% the size of the inverter, apart from some government rebates this is also recognising that in some months the solar intensity is lower and we need more panels to feed into the inverter to keep it near capacity (with batteries we can even go 150-200% ratio).

In regards your sensor, I would drop the unit of measurement (kWh/kWp) as that will cause errors in HA (check the logs they will be there).
Also the friendly name can be two spaces the left (same indent as name and unit of measurement etc), as friendly name isn't an attribute (that then adds extra issues calling it).
State class of measurement would probably be best for this also, dropping the device class of energy).

**Comment by slipx06** [comment]
@Elrikus Please test https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v4.43.1

**Reply by Elrikus** [reply]
> @Elrikus Please test https://github.com/slipx06/sunsynk-power-flow-card/releases/tag/v4.43.1

Sorry my friend, got health issues. I'll test it in a few days. Thank you for your support and I apologize for the delay. 😔

**Reply by Elrikus** [reply]
I made update and i looks like it work! Thank You!
